internal/balancergroup/balancergroup_test.go

```diff
@@ -640,3 +751,78 @@ func (s) TestBalancerGracefulSwitch(t *testing.T) {
 		}
 	}
 }
+
+func (s) TestBalancerExitIdle_All(t *testing.T) {
+	balancerOne := "stub-balancer-test-balancer-group-exit-idle-one"
+	balancerTwo := "stub-balancer-test-balancer-group-exit-idle-two"
+	balancerThree := "stub-balancer-test-balancer-group-exit-idle-three"
+
+	balancerNames := []string{balancerOne, balancerTwo, balancerThree}
+	testIDs := []string{testBalancerIDs[0], testBalancerIDs[1], testBalancerIDs[2]}
+
+	exitIdleCh := make(chan string, len(balancerNames))
+
+	for _, name := range balancerNames {
+		stub.Register(name, stub.BalancerFuncs{
+			ExitIdle: func(_ *stub.BalancerData) {
+				exitIdleCh <- name
+			},
+		})
+	}
+
+	cc := testutils.NewBalancerClientConn(t)
+	bg := New(Options{
+		CC:              cc,
+		BuildOpts:       balancer.BuildOptions{},
+		StateAggregator: nil,
+		Logger:          nil,
+	})
+	defer bg.Close()
+
+	bg.Add(testIDs[0], balancer.Get(balancerOne))
+	bg.Add(testIDs[1], balancer.Get(balancerTwo))
+	bg.Add(testIDs[2], balancer.Get(balancerThree))
+
+	bg.ExitIdle()
+
+	called := make(map[string]bool)
+	for i := 0; i < len(balancerNames); i++ {
+		select {
+		case name := <-exitIdleCh:
+			if called[name] {
+				t.Fatalf("ExitIdle was called multiple times for sub-balancer %q", name)
+			}
+			called[name] = true
+		case <-time.After(time.Second):
+			t.Fatalf("Timeout: ExitIdle not called for all sub-balancers, got %d/%d", len(called), len(balancerNames))
+		}
+	}
+}
+
+func (s) TestBalancerGroup_ExitIdle_AfterClose(t *testing.T) {
+	balancerName := "stub-balancer-test-balancer-group-exit-idle-after-close"
+	exitIdleCh := make(chan struct{})
+
+	stub.Register(balancerName, stub.BalancerFuncs{
+		ExitIdle: func(_ *stub.BalancerData) {
+			close(exitIdleCh)
+		},
+	})
+
+	bg := New(Options{
+		CC:              testutils.NewBalancerClientConn(t),
+		BuildOpts:       balancer.BuildOptions{},
+		StateAggregator: nil,
+		Logger:          nil,
+	})
+
+	bg.Add(testBalancerIDs[0], balancer.Get(balancerName))
+	bg.Close()
+	bg.ExitIdle()
+
+	select {
+	case <-exitIdleCh:
+		t.Fatalf("ExitIdle was called on sub-balancer even after BalancerGroup was closed")
+	case <-time.After(defaultTestTimeout):
```

Comment:
```suggestion
	case <-time.After(defaultShortTestTimeout):
```
We use `defaultShortTestTimeout` to wait for some time for cases that are *not* expected to happen , and use `defaultTestTimeout` for cases that are expected to happen.